ansible/deploy_vm_from_template.py

```diff
@@ -0,0 +1,237 @@
+                        help='User name to use')
+
+    parser.add_argument('--password',
+                        required=False,
+                        action='store',
+                        help='Password to use')
+
+    parser.add_argument('--datacenter',
+                        required=False,
+                        action='store',
+                        help='Datacenter to use')
+
+    parser.add_argument('--cluster',
+                        required=False,
+                        action='store',
+                        help='Cluster to use')
+
+    parser.add_argument('--template',
+                        required=True,
+                        action='store',
+                        help='Name of the template')
+
+    parser.add_argument('--folder',
+                        required=True,
+                        action='store',
+                        help='Folder in which the cloned VM will be placed')
+
+    parser.add_argument('--name',
+                        required=True,
+                        action='store',
+                        help='Name which will be given to the cloned VM')
+
+    parser.add_argument('--mgmt-network',
+                        required=True,
+                        action='store',
+                        help='Management network for VM')
+
+    parser.add_argument('--data-network',
+                        required=True,
+                        action='store',
+                        help='Data-plane network for VM')
+
+    parser.add_argument('--data-ip-address',
+                        required=True,
+                        action='store',
+                        help='Data-plane IP address')
+
+    parser.add_argument('--data-netmask',
+                        required=True,
+                        action='store',
+                        help='Data-plane netmask')
+
+    parser.add_argument('--vm-password',
+                        required=False,
+                        default=None,
+                        action='store',
+                        help='Password to set for Administrator on Windows')
+
+    args = parser.parse_args()
+
+    if not args.password:
+        args.password = getpass.getpass(
+            prompt='Enter password')
+
+    return args
+
+
+def get_vm_network_interfaces(vm=None):
+    if vm is None:
+            return []
+
+    device_list = []
+    for device in vm.config.hardware.device:
+        if isinstance(device, vim.vm.device.VirtualPCNet32) or \
+           isinstance(device, vim.vm.device.VirtualVmxnet2) or \
+           isinstance(device, vim.vm.device.VirtualVmxnet3) or \
+           isinstance(device, vim.vm.device.VirtualE1000) or \
+           isinstance(device, vim.vm.device.VirtualE1000e) or \
+           isinstance(device, vim.vm.device.VirtualSriovEthernetCard):
+            device_list.append(device)
+
+    return device_list
+
+
+def get_nic_with_updated_network(content, nic_device, network_name):
```

Comment:
Looking at the context in which this function is used, maybe calling it `get_nic_update_network_operation` would be better?